packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/staticResourcesMixin.ts

```diff
@@ -35,30 +36,94 @@ export const staticResourcesMixin = createSingletonMixin(
       baseUrlMixin,
     )(superclass);
 
+    // the logic should be as following:
+    // if there is a local storage override, use it
+    // otherwise, if there is a base-static-url attribute, use it
+    // otherwise, try to use base-url, and check if it's working
+    // if it's working, use it
+    // if not, use the default content url
     return class StaticResourcesMixinClass extends BaseClass {
+      #lastBaseUrl?: string;
+      #workingBaseUrl?: string;
+
+      #getResourceUrls(filename: string): string[] | string {
+        if (this.baseStaticUrl) {
+          return getResourceUrl({
+            projectId: this.projectId,
+            filename,
+            baseUrl: this.baseStaticUrl,
+          });
+        }
+
+        const isBaseUrlUpdated = this.#lastBaseUrl !== this.baseUrl;
+        const shouldFallbackFetch = isBaseUrlUpdated && !!this.baseUrl;
```

Comment:
is there a case where there is any empty baseUrl?

Reply:
fixed